subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestSuiteTestFrameworkIntegrationTest.groovy

```diff
@@ -0,0 +1,443 @@
+        """
+        writeJUnitJupiterSources(file("src/test/java"))
+
+        when:
+        succeeds("check")
+        then:
+        DefaultTestExecutionResult result = new DefaultTestExecutionResult(testDirectory)
+        result.assertTestClassesExecuted("com.example.IncludedTest")
+        result.assertTestClassesNotExecuted("com.example.ExcludedTest")
+    }
+
+    def "options for test framework are respected for JUnitJupiter for custom test suite"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            repositories {
+                ${mavenCentralRepository()}
+            }
+
+            testing {
+                suites {
+                    integrationTest(JvmTestSuite) {
+                        useJUnitJupiter()
+                        targets.all {
+                            testTask.configure {
+                                options {
+                                    excludeTags "exclude"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            
+            check.dependsOn testing.suites
+        """
+        writeJUnitJupiterSources(file("src/integrationTest/java"))
+
+        when:
+        succeeds("check")
+        then:
+        DefaultTestExecutionResult result = new DefaultTestExecutionResult(testDirectory, 'build', '', '', 'integrationTest')
+        result.assertTestClassesExecuted("com.example.IncludedTest")
+        result.assertTestClassesNotExecuted("com.example.ExcludedTest")
+    }
+
+    def "options for test framework are respected for JUnitJupiter for custom test suite without explicit configuration"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            repositories {
+                ${mavenCentralRepository()}
+            }
+
+            testing {
+                suites {
+                    integrationTest(JvmTestSuite) {
+                        targets.all {
+                            testTask.configure {
+                                options {
+                                    excludeTags "exclude"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            
+            check.dependsOn testing.suites
+        """
+        writeJUnitJupiterSources(file("src/integrationTest/java"))
+
+        when:
+        succeeds("check")
+        then:
+        DefaultTestExecutionResult result = new DefaultTestExecutionResult(testDirectory, 'build', '', '', 'integrationTest')
+        result.assertTestClassesExecuted("com.example.IncludedTest")
+        result.assertTestClassesNotExecuted("com.example.ExcludedTest")
+    }
+
+    def "options for test framework are respected for JUnitJupiter for custom test suite with changing test framework"() {
```

Comment:
Should there be a further test where the whole `testing` extension block is repeated and changes the configuration?

Reply:
Tom's suggestion sounds similar to this snippet: https://github.com/gradle/gradle/blob/release/subprojects/docs/src/snippets/testing/test-suite-configure-source-dir/kotlin/build.gradle.kts#L43

I agree a test with proper assertions is a good idea.